### PR TITLE
Expose name field in AccountInfo

### LINKED ISSUE #2147: AccountInfo doesn't return name of account
Please follow the issue template below. Failure to do so will result in a delay in answering your question.

## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [X] `@azure/msal-browser@2.0.1`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description

The AccountInfo interface doesn't expose a name property 
https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/7cd420254b57b3a988446dc5774659a81a2bbd2a/lib/msal-common/src/cache/entities/AccountEntity.ts#L95
It looks like this object already has the name of the account, it is just not exposed

Is there reason for this? Could this be exposed? It is useful for us to render the name of the currently logged in user 
Thanks!

### PR DESCRIPTION
**Pull request by jmckennon**
Fixes #2147

This adds the  optional `name` field to the `AccountInfo` type, pulling it directly from `AccountEntity`. This exposes it to our users through `getAllAccounts()`

### CODE REVIEW

**Review by jasonnutter — APPROVED** [review]

**Review by jo-arroyo — APPROVED** [review]

File: lib/msal-common/test/client/SilentFlowClient.spec.ts [hunk 1]
```diff
@@ -235,7 +236,8 @@ describe("SilentFlowClient unit tests", () => {
             homeAccountId: `${TEST_DATA_CLIENT_INFO.TEST_UID}.${TEST_DATA_CLIENT_INFO.TEST_UTID}`,
             tenantId: ID_TOKEN_CLAIMS.tid,
             environment: "login.windows.net",
-            username: ID_TOKEN_CLAIMS.preferred_username
+            username: ID_TOKEN_CLAIMS.preferred_username,
+            name: "Abe Lincoln"
```
Comment (tnorling) [inline lib/msal-common/test/client/SilentFlowClient.spec.ts:240]:
```suggestion
            name: ID_TOKEN_CLAIMS.name
```

Comment (tnorling) [reply lib/msal-common/test/client/SilentFlowClient.spec.ts:240]: Can we update this everywhere it was hardcoded?

File: change/@azure-msal-browser-2020-09-14-16-48-24-accountinfo-name.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "patch",
+  "comment": "Add name field to AccountInfo",
```
Comment (tnorling) [inline change/@azure-msal-browser-2020-09-14-16-48-24-accountinfo-name.json:3]: Could you add the PR number to the comments in each of these?

**Review by tnorling — APPROVED** [review]
2 comments but otherwise looks good

File: change/@azure-msal-browser-2020-09-14-16-48-24-accountinfo-name.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "patch",
```
Comment (tnorling) [inline change/@azure-msal-browser-2020-09-14-16-48-24-accountinfo-name.json:2]: Browser changes are just a test so this should be type None

**Pull request merged**